tests/integration/fixtures/split_data/test_sql_data_splitting.py

```diff
@@ -0,0 +1,255 @@
+
+    with open("connection_string.yml") as f:
+        db_config: dict = yaml_handler.load(f)
+
+    dialect: str = db_config["dialect"]
+    if dialect == "snowflake":
+        CONNECTION_STRING: str = get_snowflake_connection_url()
+    elif dialect == "bigquery":
+        CONNECTION_STRING: str = get_bigquery_connection_url()
+    else:
+        CONNECTION_STRING: str = db_config["connection_string"]
+
+    return dialect, CONNECTION_STRING
+
+
+TAXI_DATA_TABLE_NAME: str = "taxi_data_all_samples"
+
+
+def _load_data(
+    connection_string: str, table_name: str = TAXI_DATA_TABLE_NAME
+) -> pd.DataFrame:
+
+    # Load the first 10 rows of each month of taxi data
+    return load_data_into_test_database(
+        table_name=table_name,
+        csv_paths=[
+            f"./data/yellow_tripdata_sample_{year}-{month}.csv"
+            for year in ["2018", "2019", "2020"]
+            for month in [f"{mo:02d}" for mo in range(1, 12 + 1)]
+        ],
+        connection_string=connection_string,
+        convert_colnames_to_datetime=["pickup_datetime", "dropoff_datetime"],
+    )
+
+
+dialect, CONNECTION_STRING = _get_connection_string_and_dialect()
+print(f"Testing dialect: {dialect}")
+
+test_df = _load_data(connection_string=CONNECTION_STRING)
+
+
+YEARS_IN_TAXI_DATA = (
+    pd.date_range(start="2018-01-01", end="2020-12-31", freq="AS")
+    .to_pydatetime()
+    .tolist()
+)
+YEAR_BATCH_IDENTIFIER_DATA: List[dict] = [
+    {DatePart.YEAR.value: dt.year} for dt in YEARS_IN_TAXI_DATA
+]
+
+MONTHS_IN_TAXI_DATA = (
+    pd.date_range(start="2018-01-01", end="2020-12-31", freq="MS")
+    .to_pydatetime()
+    .tolist()
+)
+YEAR_MONTH_BATCH_IDENTIFIER_DATA: List[dict] = [
+    {DatePart.YEAR.value: dt.year, DatePart.MONTH.value: dt.month}
+    for dt in MONTHS_IN_TAXI_DATA
+]
+MONTH_BATCH_IDENTIFIER_DATA: List[dict] = [
+    {DatePart.MONTH.value: dt.month} for dt in MONTHS_IN_TAXI_DATA
+]
+
+TEST_COLUMN: str = "pickup_datetime"
+
+
+# Since taxi data does not contain all days, we need to introspect the data to build the fixture:
+YEAR_MONTH_DAY_BATCH_IDENTIFIER_DATA: List[dict] = list(
+    {val[0]: val[1], val[2]: val[3], val[4]: val[5]}
+    for val in {
+        (
+            DatePart.YEAR.value,
+            dt.year,
+            DatePart.MONTH.value,
+            dt.month,
+            DatePart.DAY.value,
+            dt.day,
+        )
+        for dt in test_df[TEST_COLUMN]
+    }
+)
+YEAR_MONTH_DAY_BATCH_IDENTIFIER_DATA: List[dict] = sorted(
+    YEAR_MONTH_DAY_BATCH_IDENTIFIER_DATA,
+    key=lambda x: (
+        x[DatePart.YEAR.value],
+        x[DatePart.MONTH.value],
+        x[DatePart.DAY.value],
+    ),
+)
+
+
+@dataclass
+class SqlSplittingTestCase:
+    splitter_method_name: str
+    splitter_kwargs: dict
+    num_expected_batch_definitions: int
+    num_expected_rows_in_first_batch_definition: int
+    expected_pickup_datetimes: List[dict]
+
+
+test_cases: List[SqlSplittingTestCase] = [
```

Comment:
Really great to see this pattern showing up in places other than `test_script_runner.py` :)